api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/ClaimResult.java

```diff
@@ -69,6 +74,19 @@ private static VerificationResult buildVerificationResult(CombinedIdentityAndEli
                 .postcodeMatch(identityAndEligibilityResponse.getPostcodeMatch())
                 .pregnantChildDOBMatch(identityAndEligibilityResponse.getPregnantChildDOBMatch())
                 .qualifyingBenefits(identityAndEligibilityResponse.getQualifyingBenefits())
+                .isPregnantOrAtLeast1ChildMatched(isPregnantOrAtLeastOneChildMatches(claim, identityAndEligibilityResponse.getDobOfChildrenUnder4()))
                 .build();
     }
+
+    private static Boolean isPregnantOrAtLeastOneChildMatches(Claim claim, List<LocalDate> registeredChildren) {
+        Claimant claimant = claim.getClaimant();
+        if (claimant.getExpectedDeliveryDate() != null) { // relying on validation in ClaimantDTO to ensure this is a valid value
+            return Boolean.TRUE;
+        }
+        List<LocalDate> declaredChildren = claimant.getChildrenDob();
+        if (isEmpty(declaredChildren) || isEmpty(registeredChildren)) {
```

Comment:
Is this if statement necessary? Isn't this scenario handled by `containsAny` ?

Reply:
I was playing safe, in case either collection is null